include/sys/emmc.h

```diff
@@ -0,0 +1,303 @@
+                                              uint32_t arg1, uint32_t arg2,
+                                              emmc_resp_type_t exp_rt,
+                                              emmc_app_flags_t app_flags,
+                                              emmc_response_t *resp) {
+  device_t *idev = EMMC_METHOD_PROVIDER(dev, send_app_cmd);
+  return emmc_methods(idev->parent)
+    ->send_app_cmd(dev, cmd, arg1, arg2, exp_rt, app_flags, resp);
+}
+
+/**
+ * \brief Wait until e.MMC signals
+ * \param dev e.MMC controller device
+ * \param wflags conditions to be met
+ */
+static inline int emmc_wait(device_t *dev, emmc_wait_flags_t wflags) {
+  device_t *idev = EMMC_METHOD_PROVIDER(dev, wait);
+  return emmc_methods(idev->parent)->wait(dev, wflags);
+}
+
+/**
+ * \brief Read data fetched by the controller
+ * \param dev e.MMC controller device
+ * \param buf pointer to where the data should be written to
+ * \param len expected data length
+ * \param read pointer for the number of read bytes or NULL
+ * \return EMMC_OK on success
+ */
+static inline emmc_result_t emmc_read(device_t *dev, void *buf, size_t len,
+                                      size_t *read) {
+  device_t *idev = EMMC_METHOD_PROVIDER(dev, read);
+  return emmc_methods(idev->parent)->read(dev, buf, len, read);
+}
+
+/**
+ * \brief Write data to the controller
+ * \param dev e.MMC controller device
+ * \param buf pointer to where the data should be read from
+ * \param len data length
+ * \param wrote pointer for the number of written bytes in or NULL
+ * \return EMMC_OK on success
+ */
+static inline emmc_result_t emmc_write(device_t *dev, const void *buf,
+                                       size_t len, size_t *wrote) {
+  device_t *idev = EMMC_METHOD_PROVIDER(dev, write);
+  return emmc_methods(idev->parent)->write(dev, buf, len, wrote);
+}
+
+/**
+ * \brief Geta a value describing the e.MMC device
+ * \param dev e.MMC controller device
+ * \param id value identifier
+ * \param var pointer to where the associated value should be written to
+ * \return -1 if value was fetched successfully, 0 if it wasn't
+ */
+static inline int emmc_get_prop(device_t *dev, emmc_prop_id_t id,
+                                emmc_prop_val_t *val) {
+  device_t *idev = EMMC_METHOD_PROVIDER(dev, get_prop);
+  return emmc_methods(idev->parent)->get_prop(dev, id, val);
+}
+
+/**
+ * \brief Seta a value describing the e.MMC device
+ * \param dev e.MMC controller device
+ * \param id value identifier
+ * \param var pointer to where the associated value should be written to
+ * \return -1 if value was fetched successfully, 0 if it wasn't
+ */
+static inline int emmc_set_prop(device_t *dev, emmc_prop_id_t id,
+                                emmc_prop_val_t val) {
+  device_t *idev = EMMC_METHOD_PROVIDER(dev, set_prop);
+  return emmc_methods(idev->parent)->set_prop(dev, id, val);
+}
+
+static inline emmc_device_t *emmc_device_of(device_t *device) {
```

Comment:
Why this method doesn't have comment?
Intentionally or by oversight?

Reply:
Does it really need one? Look at the body. It's easy enough to tell what it does. Besides it's similar to `pci_device_of`. It's a helper function meant to be used within drivers code.

The other functions have long descriptions simply because they are just wrappers for the interface and whoever implements the interface should follow the specification given by these comments.

Reply:
If every driver developer will know what it does it doesn't need a comment.
I don't know yet what `pci_device_of` is. But I guess if I had to develop something here I'd know?

Reply:
Yes, you would. It's a helper function and a comment would be longer than the body, which pretty much defeats its purpose. It's not like quake's inverse square root procedure, where there's some kind of an obscure magical trick that needs an in-depth explanation.